server/src/main/java/org/jboss/as/server/controller/git/GitRepository.java

```diff
@@ -82,20 +80,13 @@ public GitRepository(GitRepositoryConfiguration gitConfig)
         this.sign = gitConfig.isSign();
         File baseDir = basePath.toFile();
         File gitDir = new File(baseDir, DOT_GIT);
+        SshSessionFactory sshdSessionFactory;
         if (gitConfig.getAuthenticationConfig() != null) {
             CredentialsProvider.setDefault(new ElytronClientCredentialsProvider(gitConfig.getAuthenticationConfig()));
+            sshdSessionFactory = new ElytronClientSshdSessionFactory(gitConfig.getAuthenticationConfig());
+        } else {
+            sshdSessionFactory = new SshdSessionFactory();
```

Comment:
This line just ensures that if users don't configure an authentication file it will still use the apache mina factory for SSH connections and not the default JSch factory. Users would be able to establish an SSH connection if they had all their keys and known hosts files in the default location and with the default names

Reply:
Instead of defaulting to ```SshdSessionFactory``` here (which allows an additional algorithm to be used that we aren't supporting yet), I'm wondering if we could modify ```ElytronClientSshdSessionFactory```so that if ```gitConfig.getAuthenticationConfig()``` is ```null```, it just sets our custom defaults and then just delegates to the parent methods.

Reply:
Fixed, thanks!